org-gcal.el

```diff
@@ -2282,6 +2160,24 @@ non-nil."
     (plist-get plst :year))))
 
 
+(defun org-gcal-reload-client-id-secret ()
+  "Setup OAuth2 authentication after setting client ID and secret."
+  (interactive)
+  (add-to-list
+   'oauth2-auto-additional-providers-alist
```

Comment:
Ideally you'd just be able to use `'google` and let `oauth2-auto` handle everything. I understand that this way it is possible to use the existing `org-gcal-client-id` and `org-gcal-client-secret`, but if Google verifies `oauth2-auto`, it's going to be much easier to just use its ID and secret. One more thing not to worry about :)

Reply:
Does it still work to use one secret for all clients of the library?

Reply:
Do you mean: if more than one program uses the library, can they all use the same secret?

I think so, and it's also nicer for the user because they only have to log in once. We'll have to verify scopes for all the possible libraries on that secret though.

For existing users of `org-gcal` maybe it's better to do it like you have done it now, though. This way they don't have to modify the config.

Does this answer your q?